app/features/text_rewriter/Dockerfile

```diff
@@ -0,0 +1,31 @@
+# Use Python 3.12 slim image
+FROM python:3.12-slim
+
+# Set working directory inside the container
+WORKDIR /app
+
+# Copy requirements.txt and install dependencies
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r requirements.txt \
+    && rm -rf /root/.cache
+    
+# Copy the .json service account file into the container
+COPY gen-lang-client-0717318025-ef6f992883f8.json /app/gen-lang-client-0717318025-ef6f992883f8.json
+
+# Set the environment variable for Google Cloud authentication
+ENV GOOGLE_APPLICATION_CREDENTIALS=/app/gen-lang-client-0717318025-ef6f992883f8.json
```

Comment:
I used this because I had trouble using langchain and could only get the model to work via google gemini and a vertex ai API key. With the examples you provided I'm rewriting the code to get rid of the vertext ai API.